### PR TITLE
added solr startup check. Refactored to make props available

### PR DESCRIPTION
**Pull request by artntek**
* added solr startup check. 
* Also refactored to make properties available for this solr check (and for future checks). 
* Added tests

### CODE REVIEW

File: src/edu/ucsb/nceas/metacat/healthchecks/StartupRequirementsListener.java [hunk 1]
```diff
@@ -85,24 +96,28 @@ public void contextDestroyed(ServletContextEvent sce) {
...
-            metacatProperties.load(Files.newBufferedReader(defaultPropsFilePath));
-            metacatProperties.store(Files.newBufferedWriter(defaultPropsFilePath), "");
-
-            sitePropsFilePath = Paths.get(
-                metacatProperties.getProperty(PropertyService.SITE_PROPERTIES_DIR_PATH_KEY),
-                PropertyService.SITE_PROPERTIES_FILENAME);
-
+            // can read?
+            defaultProperties.load(Files.newBufferedReader(defaultPropsFilePath));
+            // can write? Use isWriteable() so we don't mess up props file formatting for Jing :-)
```
Comment (taojing2002) [inline src/edu/ucsb/nceas/metacat/healthchecks/StartupRequirementsListener.java:112]: Thanks. But the metacat.properties file is still changed  :).  I believe it happens when we write some properties (such as status) into the file.

File: src/edu/ucsb/nceas/metacat/healthchecks/StartupRequirementsListener.java [hunk 2]
```diff
@@ -228,21 +297,23 @@ protected void abort(String message, Exception e) throws RuntimeException {
...
      * @throws RuntimeException if any unrecoverable problems are found that should cause startup
      *                          to be aborted
      */
-    private void validateSitePropertiesFileRwAccess(Path sitePropsFilePath)
-        throws RuntimeException {
+    private void validateSitePropertiesFileRwAccess(
+        Path sitePropsFilePath, Properties defaultProperties) throws RuntimeException {
 
-        Properties siteProperties = new Properties();
+        runtimeProperties = new Properties(defaultProperties);
```
Comment (taojing2002) [inline src/edu/ucsb/nceas/metacat/healthchecks/StartupRequirementsListener.java:311]: Is there any reason why we need the `defaultProperties` to initialize the runtimeProperties? If we just use the default constructor to initialize the runtimeProperties object, we can only has one parameter in the method signature.

File: src/edu/ucsb/nceas/metacat/healthchecks/StartupRequirementsListener.java (hunk 1 above)
Comment (artntek) [reply src/edu/ucsb/nceas/metacat/healthchecks/StartupRequirementsListener.java:112]: agreed - I just thought I would be better not to have it happen in yet another place :-)

**Review by taojing2002 — APPROVED** [review]
Looks good. Thank you!

**Pull request merged**